src/lib/y2storage/proposal/autoinst_partitioner.rb

```diff
@@ -0,0 +1,164 @@
+# encoding: utf-8
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/planned"
+require "y2storage/proposal/creator_result"
+require "y2storage/proposal/autoinst_partition_size"
+
+module Y2Storage
+  module Proposal
+    # Class to create and reuse partitions in any given devicegraph, based on
+    # planned devices with an AutoYaST specification of the sizes.
+    class AutoinstPartitioner
+      include Yast::Logger
+      include AutoinstPartitionSize
+
+      # Constructor
+      #
+      # @param devicegraph [Devicegraph] see {#devicegraph}
+      def initialize(devicegraph)
+        @devicegraph = devicegraph
+      end
+
+      # Reuses partitions in the target devicegraph
+      #
```

Comment:
Please, add a note to make clear that the `devicegraph` is modified (given that's the case).